Src/Support/Google.Apis.Auth/OAuth2/ServiceAccountCredential.cs

```diff
@@ -150,28 +151,34 @@ public Initializer FromCertificate(X509Certificate2 certificate)
         /// Gets the email address of the user the application is trying to impersonate in the service account flow 
         /// or <c>null</c>.
         /// </summary>
-        public string User { get { return user; } }
+        public string User { get; private set; }
 
         /// <summary>Gets the service account scopes.</summary>
-        public IEnumerable<string> Scopes { get { return scopes; } }
+        public IEnumerable<string> Scopes { get; private set; }
 
         /// <summary>
         /// Gets the key which is used to sign the request, as specified in
         /// https://developers.google.com/accounts/docs/OAuth2ServiceAccount#computingsignature.
         /// </summary>
-        public RsaKey Key { get { return key; } }
+        public RsaKey Key { get; private set; }
+
+        /// <summary>
+        /// Gets the key id of the key which is used to sign the request.
+        /// </summary>
+        public string KeyId { get; }
 
         /// <summary><c>true</c> if this credential has any scopes associated with it.</summary>
-        internal bool HasScopes { get { return scopes != null && scopes.Any(); } }
+        internal bool HasScopes { get { return Scopes != null && Scopes.Any(); } }
```

Comment:
Use an expression-bodied property while we're making the change? (And maybe the null-forgiving operator - `internal bool HasScopes => Scopes?.Any() == true;`